UCT/IB: Fix cq creation failure using legacy ibv api

dsharma283 opened the pull request:
With the rdma-core version 28.0 or higher the error code
returned by libibverbs for the case where ibv_cq_ex_to_cq
is not supported by the underlying transport has changed.
The transport layer now returns EOPNOTSUPP instead of 
ENOSYS.
Ucx is broken for all the transport devices which do not
have ibv_cq_ex_to_cq implemented. Fixing the error code
check in uct for IB devices.

Signed-off-by: Devesh Sharma <devesh.sharma@broadcom.com>

swx-jenkins3 commented:
Can one of the admins verify this patch?

brminich commented:
ok to test

brminich commented:
@dsharma283, can you pleasse fix commit title according to https://github.com/openucx/ucx/wiki/Guidance-for-contributors#general-guidelines?
(seems need to use upper case for `UCT`)

dsharma283 commented:
> 
> 
> @dsharma283, can you pleasse fix commit title according to https://github.com/openucx/ucx/wiki/Guidance-for-contributors#general-guidelines?
> (seems need to use upper case for `UCT`)

Updated the patch subject line. Added one more change to make it backwards compatible.

yosefe commented:
/azp run

yosefe commented:
/azp run

shamisp commented:
@dsharma283 - Thanks for the patch ! Please add Broadcom copyright to the modified files. In order to merge the PR we need signed CLA: https://www.openucx.org/wp-content/uploads/2019/05/ucx-corporate-contributor-agrement.pdf  . Once it is signed, please send the copy to me. Thanks !

shamisp reviewed (changes requested):
Pending - waiting for CLA

yosefe commented:
/azp run

yosefe commented:
/azp run

dsharma283 commented:
> @dsharma283 - Thanks for the patch ! Please add Broadcom copyright to the modified files. In order to merge the PR we need signed CLA: https://www.openucx.org/wp-content/uploads/2019/05/ucx-corporate-contributor-agrement.pdf . Once it is signed, please send the copy to me. Thanks !

Thanks Pasha for this note, let me take it to my Org and get back to you with approvals.

dsharma283 commented:
I am still taking time to come back on CLA. This may take some more time

dsharma283 commented:
@shamisp Hi, I have received a signed copy of CLA, Please let me know where I should send the Signed CLA. I will add the broadcom copyright to the modified files.

shamisp commented:
Please email to admin at ucfconsortium.org . Once it is sent, please let me know and I will make sure that it is counter signed.

dsharma283 commented:
@shamisp Sent!

shamisp commented:
The CLA was received.

shamisp commented:
@yosefe FYI

dmitrygx commented on src/uct/ib/base/ib_iface.c:
```diff
@@ -952,7 +952,7 @@ ucs_status_t uct_ib_verbs_create_cq(uct_ib_iface_t *iface, uct_ib_dir_t dir,
 
     cq = ibv_cq_ex_to_cq(ibv_create_cq_ex(dev->ibv_context, &cq_attr));
-    if (!cq && (errno == ENOSYS))
+    if (!cq && (errno == EOPNOTSUPP || errno == ENOSYS))
```
pls, warp each condition by `()`:
```
if (!cq && ((errno == EOPNOTSUPP) || (errno == ENOSYS)))
```

dsharma283 replied:
Done!

shamisp commented:
Please add Broadcom copyright to the header of the file.

shamisp commented:
@yosefe I think this has to be back ported to 1.10.x ?

yosefe commented:
> @yosefe I think this has to be back ported to 1.10.x ?

it's up to @dsharma283

dsharma283 commented:
> 
> 
> > @yosefe I think this has to be back ported to 1.10.x ?
> 
> it's up to @dsharma283

I can do that, it is good to have.

shamisp commented:
thanks, please open PR for v1.10.x branch. Much appreciated.

shamisp commented:
@dmitrygx Is it good to go? thanks !
